python/ray/train/callbacks/logging.py

```diff
@@ -174,6 +175,100 @@ def _validate_worker_to_log(self, worker_to_log) -> int:
+        registry_uri (Optional[str]): The registry URI that gets passed
+            directly to mlflow initialization.
+        experiment_id (Optional[str]): The experiment id of an already
+            existing experiment. If not
+            passed in, experiment_name will be used.
+        experiment_name (Optional[str]): The experiment name to use for this
+            Train run.
+            If the experiment with the name already exists with MLflow,
+            it will be used. If not, a new experiment will be created with
+            this name.
+        tags (Optional[Dict]):  An optional dictionary of string keys and
+            values to set as tags on the run
+        save_artifact (bool): If set to True, automatically save the entire
+            contents of the Train local_dir as an artifact to the
+            corresponding run in MlFlow.
+        logdir (Optional[str]): Path to directory where the results file
+            should be. If None, will be set by the Trainer. If no tracking
+            uri or registry uri are passed in, the logdir will be used for
+            both.
+        worker_to_log (int): Worker index to log. By default, will log the
+            worker with index 0.
+    """
+
+    def __init__(self,
+                 tracking_uri: Optional[str] = None,
+                 registry_uri: Optional[str] = None,
+                 experiment_id: Optional[str] = None,
+                 experiment_name: Optional[str] = None,
+                 tags: Optional[Dict] = None,
+                 save_artifact: bool = False,
+                 logdir: Optional[str] = None,
+                 worker_to_log: int = 0):
+        super().__init__(logdir=logdir, worker_to_log=worker_to_log)
+
+        self.tracking_uri = tracking_uri
+        self.registry_uri = registry_uri
+        self.experiment_id = experiment_id
+        self.experiment_name = experiment_name
+        self.tags = tags
+
+        self.save_artifact = save_artifact
+        self.mlflow_util = MLflowLoggerUtil()
+
+    def start_training(self, logdir: str, config: Dict, **info):
+        super().start_training(logdir=logdir, config=config, info=info)
+
+        tracking_uri = self.tracking_uri if self.tracking_uri is not None \
+            else \
+            str(self.logdir)
+        registry_uri = self.registry_uri if self.registry_uri is not None \
+            else \
+            str(self.logdir)
+
+        success = self.mlflow_util.setup_mlflow(
+            tracking_uri=tracking_uri,
+            registry_uri=registry_uri,
+            experiment_id=self.experiment_id,
+            experiment_name=self.experiment_name,
+            create_experiment_if_not_exists=True)
+
+        if not success:
+            raise ValueError("No experiment_name or experiment_id passed in, "
+                             "Please "
+                             "set one of these to use the "
+                             "MLflowLoggerCallback.")
```

Comment:
I'm wondering if you can instead just change `setup_mlflow` to not return anything, and raise an Exception in the case of a failure. 

It seems a little strange to me that `setup_mlflow` returns a `Boolean` that captures if it succeeds or fails, while here we'll process it as a specific error type.

Reply:
Originally this was done so each specific logger can have a custom error message. But good point, this can be consolidated into one. Made the change!